kratos/geometries/geometry.h

```diff
@@ -647,9 +647,120 @@ class Geometry
     ///@name Operations
     ///@{
 
-    virtual Pointer Create( PointsArrayType const& ThisPoints ) const
+    /**
+     * @brief Creates a new geometry pointer
+     * @param rThisPoints the nodes of the new geometry
+     * @return Pointer to the new geometry
+     */
+    virtual Pointer Create(
+        PointsArrayType const& rThisPoints
+    ) const
+    {
+        // Create geometry
+        auto p_geom = this->Create(0, rThisPoints);
+
+        // Generate Id
+        IndexType id = reinterpret_cast<IndexType>(p_geom.get());
+
+        // Sets second bit to zero.
+        p_geom->SetIdSelfAssigned(id);
+
+        // Sets first bit to zero.
+        p_geom->SetIdNotGeneratedFromString(id);
+
+        // Sets Id
+        p_geom->SetIdWithoutCheck(id);
+
+        return p_geom;
+    }
+
+    /**
+     * @brief Creates a new geometry pointer
+     * @param NewGeometryId the ID of the new geometry
+     * @param rThisPoints the nodes of the new geometry
+     * @return Pointer to the new geometry
+     */
+    virtual Pointer Create(
+        const IndexType NewGeometryId,
+        PointsArrayType const& rThisPoints
+    ) const
+    {
+        return Pointer( new Geometry( NewGeometryId, rThisPoints, mpGeometryData));
+    }
+
+    /**
+     * @brief Creates a new geometry pointer
+     * @param rNewGeometryName the name of the new geometry
+     * @param rThisPoints the nodes of the new geometry
+     * @return Pointer to the new geometry
+     */
+    virtual Pointer Create(
+        const std::string& rNewGeometryName,
+        PointsArrayType const& rThisPoints
+        ) const
+    {
+        auto p_geom = this->Create(0, rThisPoints);
+        p_geom->SetId(rNewGeometryName);
+        return p_geom;
+    }
+    
+    /**
+     * @brief Creates a new geometry pointer
+     * @param pGeometry Pointer to an existing geometry
+     * @return Pointer to the new geometry
+     */
+    virtual Pointer Create(
+        Pointer pGeometry
+    ) const
     {
-        return Pointer( new Geometry( ThisPoints, mpGeometryData ) );
+        // Create geometry
+        auto p_geom = this->Create(0, pGeometry);
```

Comment:
BTW, what would be the use case for this? Shouldn't be `Clone`?

I mean, if we need a clone of existing geometry, so let's call it `Clone`. If we only need the array of points then passing geometry to the `Create(PointsArrayType)` would work because  geometry is already an array of points.

So should it be a `Clone` or there is another use case justifying this one?

Reply:
I chose Create because I thought it was the proper name, but can be renamed as Clone if necessary

Reply:
But do we need it? Please note that then we should have to define it in all derived geometries.

Reply:
I found an answer to my concern. It should be Create as the id is regenerated for the new geometry. (Clone would maintain the id)

Reply:
And the difference to the PointsArrayType is the data container that is also copied.